### PR TITLE
dts: arm: nxp: lpc55s1x: add usbphy1 devicetree node

### LINKED ISSUE #74605: boards: nxp: lpcxpress55s16: missing usbphy1 devicetree node
**Describe the bug**
A devicetree node for the usbphy1 present on the NXP LPC55S1x series was left out of commit 0c70a72ac7048376e20cef433335e1ce796b70dd, leading to compilation errors for `drivers/usb/udc/udc_mcux_ip3511.c` for the `lpcxpresso55s16` board.

**To Reproduce**
1. `west build -b lpcxpresso55s16/lpc55s16 samples/subsys/usb/hid-keyboard/`
2. See error

**Expected behavior**
No compilation error.

**Impact**
Unable to use `drivers/usb/udc/udc_mcux_ip3511.c` on LPC55S1x-based platforms without manually adding the needed usbphy1 node.

**Logs and console output**
```
...
[ 77%] Building C object zephyr/drivers/usb/udc/CMakeFiles/drivers__usb__udc.dir/udc_mcux_ip3511.c.obj
In file included from /home/brix/Projects/zephyrproject/zephyr/include/zephyr/sys/util_macro.h:34,
                 from /home/brix/Projects/zephyrproject/zephyr/include/zephyr/sys/util.h:17,
                 from /home/brix/Projects/zephyrproject/modules/hal/nxp/mcux/mcux-sdk/drivers/common/fsl_common.h:233,
                 from /home/brix/Projects/zephyrproject/zephyr/soc/nxp/lpc/lpc55xxx/./soc.h:19,
                 from /home/brix/Projects/zephyrproject/zephyr/drivers/usb/udc/udc_mcux_ip3511.c:8:
/home/brix/Projects/zephyrproject/zephyr/drivers/usb/udc/udc_mcux_ip3511.c:874:48: error: expected expression before ',' token
  874 |                 UDC_MCUX_PHY_CFG_PTR_OR_NULL(n),                                        \
      |                                                ^
/home/brix/Projects/zephyrproject/zephyr/include/zephyr/sys/util_internal.h:72:26: note: in definition of macro '__DEBRACKET'
   72 | #define __DEBRACKET(...) __VA_ARGS__
      |                          ^~~~~~~~~~~
/home/brix/Projects/zephyrproject/zephyr/include/zephyr/sys/util_internal.h:64:9: note: in expansion of macro '__GET_ARG2_DEBRACKET'
   64 |         __GET_ARG2_DEBRACKET(one_or_two_args _if_code, _else_code)
      |         ^~~~~~~~~~~~~~~~~~~~
/home/brix/Projects/zephyrproject/zephyr/include/zephyr/sys/util_internal.h:59:9: note: in expansion of macro '__COND_CODE'
   59 |         __COND_CODE(_XXXX##_flag, _if_1_code, _else_code)
      |         ^~~~~~~~~~~
/home/brix/Projects/zephyrproject/zephyr/include/zephyr/sys/util_macro.h:180:9: note: in expansion of macro 'Z_COND_CODE_1'
  180 |         Z_COND_CODE_1(_flag, _if_1_code, _else_code)
      |         ^~~~~~~~~~~~~
/home/brix/Projects/zephyrproject/zephyr/include/zephyr/devicetree.h:4528:9: note: in expansion of macro 'COND_CODE_1'
 4528 |         COND_CODE_1(DT_HAS_COMPAT_STATUS_OKAY(DT_DRV_COMPAT),   \
      |         ^~~~~~~~~~~
/home/brix/Projects/zephyrproject/zephyr/build/zephyr/include/generated/zephyr/devicetree_generated.h:7814:48: note: in expansion of macro 'USB_MCUX_IP3511_DEVICE_DEFINE'
 7814 | #define DT_FOREACH_OKAY_INST_nxp_lpcip3511(fn) fn(0)
      |                                                ^~
/home/brix/Projects/zephyrproject/zephyr/include/zephyr/sys/util_internal.h:105:36: note: in expansion of macro 'DT_FOREACH_OKAY_INST_nxp_lpcip3511'
  105 | #define UTIL_PRIMITIVE_CAT(a, ...) a##__VA_ARGS__
      |                                    ^
/home/brix/Projects/zephyrproject/zephyr/drivers/usb/udc/udc_mcux_ip3511.c:890:1: note: in expansion of macro 'DT_INST_FOREACH_STATUS_OKAY'
  890 | DT_INST_FOREACH_STATUS_OKAY(USB_MCUX_IP3511_DEVICE_DEFINE)
      | ^~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/brix/Projects/zephyrproject/zephyr/drivers/usb/udc/udc_mcux_ip3511.c:874:48: warning: excess elements in struct initializer
  874 |                 UDC_MCUX_PHY_CFG_PTR_OR_NULL(n),                                        \
      |                                                ^
/home/brix/Projects/zephyrproject/zephyr/include/zephyr/sys/util_internal.h:72:26: note: in definition of macro '__DEBRACKET'
   72 | #define __DEBRACKET(...) __VA_ARGS__
      |                          ^~~~~~~~~~~
/home/brix/Projects/zephyrproject/zephyr/include/zephyr/sys/util_internal.h:64:9: note: in expansion of macro '__GET_ARG2_DEBRACKET'
   64 |         __GET_ARG2_DEBRACKET(one_or_two_args _if_code, _else_code)
      |         ^~~~~~~~~~~~~~~~~~~~
/home/brix/Projects/zephyrproject/zephyr/include/zephyr/sys/util_internal.h:59:9: note: in expansion of macro '__COND_CODE'
   59 |         __COND_CODE(_XXXX##_flag, _if_1_code, _else_code)
      |         ^~~~~~~~~~~
/home/brix/Projects/zephyrproject/zephyr/include/zephyr/sys/util_macro.h:180:9: note: in expansion of macro 'Z_COND_CODE_1'
  180 |         Z_COND_CODE_1(_flag, _if_1_code, _else_code)
      |         ^~~~~~~~~~~~~
/home/brix/Projects/zephyrproject/zephyr/include/zephyr/devicetree.h:4528:9: note: in expansion of macro 'COND_CODE_1'
 4528 |         COND_CODE_1(DT_HAS_COMPAT_STATUS_OKAY(DT_DRV_COMPAT),   \
      |         ^~~~~~~~~~~
/home/brix/Projects/zephyrproject/zephyr/build/zephyr/include/generated/zephyr/devicetree_generated.h:7814:48: note: in expansion of macro 'USB_MCUX_IP3511_DEVICE_DEFINE'
 7814 | #define DT_FOREACH_OKAY_INST_nxp_lpcip3511(fn) fn(0)
      |                                                ^~
/home/brix/Projects/zephyrproject/zephyr/include/zephyr/sys/util_internal.h:105:36: note: in expansion of macro 'DT_FOREACH_OKAY_INST_nxp_lpcip3511'
  105 | #define UTIL_PRIMITIVE_CAT(a, ...) a##__VA_ARGS__
      |                                    ^
/home/brix/Projects/zephyrproject/zephyr/drivers/usb/udc/udc_mcux_ip3511.c:890:1: note: in expansion of macro 'DT_INST_FOREACH_STATUS_OKAY'
  890 | DT_INST_FOREACH_STATUS_OKAY(USB_MCUX_IP3511_DEVICE_DEFINE)
      | ^~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/brix/Projects/zephyrproject/zephyr/drivers/usb/udc/udc_mcux_ip3511.c:874:48: note: (near initialization for 'priv_config_0')
  874 |                 UDC_MCUX_PHY_CFG_PTR_OR_NULL(n),                                        \
      |                                                ^
/home/brix/Projects/zephyrproject/zephyr/include/zephyr/sys/util_internal.h:72:26: note: in definition of macro '__DEBRACKET'
   72 | #define __DEBRACKET(...) __VA_ARGS__
      |                          ^~~~~~~~~~~
/home/brix/Projects/zephyrproject/zephyr/include/zephyr/sys/util_internal.h:64:9: note: in expansion of macro '__GET_ARG2_DEBRACKET'
   64 |         __GET_ARG2_DEBRACKET(one_or_two_args _if_code, _else_code)
      |         ^~~~~~~~~~~~~~~~~~~~
/home/brix/Projects/zephyrproject/zephyr/include/zephyr/sys/util_internal.h:59:9: note: in expansion of macro '__COND_CODE'
   59 |         __COND_CODE(_XXXX##_flag, _if_1_code, _else_code)
      |         ^~~~~~~~~~~
/home/brix/Projects/zephyrproject/zephyr/include/zephyr/sys/util_macro.h:180:9: note: in expansion of macro 'Z_COND_CODE_1'
  180 |         Z_COND_CODE_1(_flag, _if_1_code, _else_code)
      |         ^~~~~~~~~~~~~
/home/brix/Projects/zephyrproject/zephyr/include/zephyr/devicetree.h:4528:9: note: in expansion of macro 'COND_CODE_1'
 4528 |         COND_CODE_1(DT_HAS_COMPAT_STATUS_OKAY(DT_DRV_COMPAT),   \
      |         ^~~~~~~~~~~
/home/brix/Projects/zephyrproject/zephyr/build/zephyr/include/generated/zephyr/devicetree_generated.h:7814:48: note: in expansion of macro 'USB_MCUX_IP3511_DEVICE_DEFINE'
 7814 | #define DT_FOREACH_OKAY_INST_nxp_lpcip3511(fn) fn(0)
      |                                                ^~
/home/brix/Projects/zephyrproject/zephyr/include/zephyr/sys/util_internal.h:105:36: note: in expansion of macro 'DT_FOREACH_OKAY_INST_nxp_lpcip3511'
  105 | #define UTIL_PRIMITIVE_CAT(a, ...) a##__VA_ARGS__
      |                                    ^
/home/brix/Projects/zephyrproject/zephyr/drivers/usb/udc/udc_mcux_ip3511.c:890:1: note: in expansion of macro 'DT_INST_FOREACH_STATUS_OKAY'
  890 | DT_INST_FOREACH_STATUS_OKAY(USB_MCUX_IP3511_DEVICE_DEFINE)
      | ^~~~~~~~~~~~~~~~~~~~~~~~~~~
gmake[2]: *** [zephyr/drivers/usb/udc/CMakeFiles/drivers__usb__udc.dir/build.make:90: zephyr/drivers/usb/udc/CMakeFiles/drivers__usb__udc.dir/udc_mcux_ip3511.c.obj] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:3922: zephyr/drivers/usb/udc/CMakeFiles/drivers__usb__udc.dir/all] Error 2
gmake: *** [Makefile:91: all] Error 2
...
```

**Environment (please complete the following information):**
 - OS: Linux
 - Toolchain: Zephyr SDK
 - Commit SHA: e96cfe04789ee964c91a913801c9c4693d5a771f

### PR DESCRIPTION
**Pull request by henrikbrixandersen**
Add devicetree node for the USB1 High Speed PHY present in the NXP LPC55S1x
series.

Fixes: #74605

Signed-off-by: Henrik Brix Andersen <henrik@brixandersen.dk>

### CODE REVIEW

**Comment by henrikbrixandersen** [comment]
The LPCXpresso55S16 was left out of commit 0c70a72ac7048376e20cef433335e1ce796b70dd.

**Comment by aescolar** [comment]
Bulk changing the milestone from everything which did not make it to the v3.7.0 freeze deadline
and which is not labeled as a bug or documentation (or does not appear clearly as only containing
a bug fix or a documentation change).
If you disagree about this change please add the milestone again.

Please note that until rc2 only PRs containing bug fixes, docs and tests can be merged.
After that and until release only bug fixes and docs changes.

Exceptions require approval by the release team and a vote by the TSC.
https://docs.zephyrproject.org/latest/project/release_process.html#stabilization-phase

**Comment by henrikbrixandersen** [comment]
@aescolar This is a stabilization fix. I'd like to see it considered for v3.7.0.

**Comment by aescolar** [comment]
> @aescolar This is a stabilization fix. I'd like to see it considered for v3.7.0.

If so, please label if as bug so it appears appropriately when searching

**Review by decsny — APPROVED** [review]

**Review by dleach02 — APPROVED** [review]

**Pull request merged**
